.azure-pipelines/windows-visual-studio.yml

```diff
@@ -0,0 +1,84 @@
+      set DMD_TESTSUITE_MAKE_ARGS=-j3
+      set DM_MAKE=%DMD_DIR%\dm\path\make.exe
+      set LDC_DIR=%DMD_DIR%\ldc2-%LDC_VERSION%-windows-multilib
+
+      FOR /F "tokens=* USEBACKQ" %%F IN (`where cl.exe`) DO (SET MSVC_CC=%%~fsF)
+      FOR /F "tokens=* USEBACKQ" %%F IN (`where lib.exe`) DO (SET MSVC_AR=%%~fsF)
+      REM this returns two lines (GNU's link.exe is on the second line)
+      REM Just take the first one
+      FOR /F "tokens=* USEBACKQ" %%F IN (`where link.exe`) DO (SET MSVC_LD=%%~fsF
+        GOTO :Next)
+      :Next
+      FOR /F "tokens=* USEBACKQ" %%F IN (`where make.exe`) DO (SET GNU_MAKE=%%~fsF)
+
+      REM WORKAROUND: move files to a directory without spaces
+      copy "%MSVC_AR%" "%DMD_DIR%\dm\path\lib.exe"
+      copy "%MSVC_LD%" "%DMD_DIR%\dm\path\link.exe"
+      copy "%MSVC_CC%" "%DMD_DIR%\dm\path\cl.exe"
+      set MSVC_AR=%DMD_DIR%\dm\path\lib.exe
+      set MSVC_CC=%DMD_DIR%\dm\path\cl.exe
+      set MSVC_LD=%DMD_DIR%\dm\path\link.exe
+
+      REM expose dm_make as default make
+      set PATH=%DMD_DIR%\dm\path;%DMD_DIR%\tools;%PATH%
+      dir "%DMD_DIR%\tools"
+      "%DMD_DIR%\tools\grep.exe" --version
+      echo %PATH%
+      grep --version
+
+      .\%VISUALD_INSTALLER% /S
+      REM configure DMD path
+      REM reg add "HKLM\SOFTWARE\DMD" /v InstallationFolder /t REG_SZ /d "%DMD_DIR%\dmd2" /reg:32 /f
+      REM configure LDC path
+      reg add "HKLM\SOFTWARE\LDC" /v InstallationFolder /t REG_SZ /d "%LDC_DIR%" /reg:32 /f
+
+      REM build via VS projects with LDC
+      cd src
+      set LDC_ARGS=%LDC_ARGS% /p:DCompiler=LDC
+      msbuild /target:dmd /p:Configuration=Release /p:Platform=Win32 %LDC_ARGS% vcbuild\dmd.sln || exit /B %ERRORLEVEL%
+
+      %DMD% --version
+      grep --version
+
+      REM Check: run druntime unittests
+      cd "%DMD_DIR%\..\druntime"
+      "%DM_MAKE%" -f win64.mak MODEL=32mscoff "DMD=%DMD%" "VCDIR=%VCINSTALLDIR%." "CC=%MSVC_CC%" "MAKE=%DM_MAKE%" target || exit /B %ERRORLEVEL%
+      echo "[DRUNTIME] running tests..."
+      "%DM_MAKE%" -f win64.mak MODEL=32mscoff "DMD=%DMD%" "VCDIR=%VCINSTALLDIR%." "CC=%MSVC_CC%" "MAKE=%DM_MAKE%" unittest32mscoff || exit /B %ERRORLEVEL%
+      "%DM_MAKE%" -f win64.mak MODEL=32mscoff "DMD=%DMD%" "VCDIR=%VCINSTALLDIR%." "CC=%MSVC_CC%" "MAKE=%DM_MAKE%" test_all || exit /B %ERRORLEVEL%
+
+      REM Check: build phobos
+      cd "%DMD_DIR%\..\phobos"
+      "%DM_MAKE%" -f win64.mak MODEL=32mscoff "DMD=%DMD%" "VCDIR=%VCINSTALLDIR%." "CC=%MSVC_CC%" "AR=%MSVC_AR%" "MAKE=%DM_MAKE%" || exit /B %ERRORLEVEL%
+
+      REM Build DMD VERSION + string imports (not built by VisualD)
+      copy "%DMD_DIR%\VERSION" "%DMD_DIR%\generated\Windows\Release\Win32\VERSION"
+
+      REM Run DMD testsuite
+      cd "%DMD_DIR%\test"
+      cp %DMD_DIR%\..\phobos\phobos32mscoff.lib .
+      "%GNU_MAKE%" -j%NUMBER_OF_PROCESSORS% all MODEL=32mscoff ARGS="-O -inline -g" OS=win32 DMD="%DMD%" "CC=%MSVC_CC%" DMD_MODEL=win32 || exit /B %ERRORLEVEL%
```

Comment:
I'm assuming your copying it because of this as well:

https://github.com/dlang/dmd/blob/master/.azure-pipelines/windows.sh#L148

How were you setting the LIB variable? As it contains the C runtime for windows, you'd have to append it like you would PATH `make LIB="$DMD_DIR/../phobos/;$LIB" ...`